packages/login-ui/components/ConfirmMnemonic/ConfirmMnemonic.js

```diff
@@ -12,84 +12,41 @@ import Button from 'components/common/ui/Button/Button'
 import {
   FORM_CONFIRM_MNEMONIC,
 } from '@chronobank/login-ui/redux/constants'
+import MnemonicButton from './MnemonicButton'
 
 import './ConfirmMnemonic.scss'
 
 class ConfirmMnemonic extends Component {
   static propTypes = {
-    mnemonic: PropTypes.string,
-    previousPage: PropTypes.func,
+    confirmPhrase: PropTypes.arrayOf(PropTypes.string),
+    currentWordsArray: PropTypes.arrayOf(PropTypes.object),
+    error: PropTypes.string,
     handleSubmit: PropTypes.func,
-    change: PropTypes.func,
-  }
-
-  static defaultProps = {
-    mnemonic: '',
-  }
-
-  constructor (props) {
-    super(props)
-
-    const wordsArray = props.mnemonic ?
-      props.mnemonic.split(' ').map((word, index) => {
-        return { index, word }
-      }) : []
-
-    this.state = {
-      confirmPhrase: [],
-      currentWordsArray: wordsArray.sort((a,b) => a.word < b.word),
-    }
-  }
-
-  handleClickWord = (word) => {
-    const { change } = this.props
-
-    if (!this.state.confirmPhrase.includes(word)) {
-      this.setState(
-        { confirmPhrase: this.state.confirmPhrase.concat(word) },
-        () => change('mnemonic', this.getCurrentMnemonic())
-      )
-    }
-  }
-
-  getCurrentMnemonic () {
-    return this.state.confirmPhrase.map((item) => item.word).join(' ')
-  }
-
-  getWordsButtons () {
-    return this.state.currentWordsArray.map((item, index) => {
-      const wordSelected = this.state.confirmPhrase.includes(item)
-
-      return (
-        <Button
-          key={index}
-          onClick={this.handleClickWord(item)}
-          styleName='word'
-          disabled={wordSelected}
-        >
-          { item.word }
-        </Button>
-      )}
-    )
-  }
-
-  clearMnemonic = () => {
-    const { change } = this.props
-
-    this.setState(
-      { confirmPhrase: [] },
-      () => change('mnemonic', this.getCurrentMnemonic())
-    )
+    onClearLastWord: PropTypes.func,
+    onClearMnemonic: PropTypes.func,
+    onClickWord: PropTypes.func,
+    previousPage: PropTypes.func,
   }
 
-  clearLastWord = () => {
-    const { change } = this.props
-
-    this.setState(
-      { confirmPhrase: this.state.confirmPhrase.slice(0, -1) },
-      () => change('mnemonic', this.getCurrentMnemonic())
-    )
-  }
+  getWordsButtons = () =>
+    this.props.currentWordsArray
+      .map((item) => {
+        const wordSelected = this.props.confirmPhrase.includes(item.word)
+        const keyIndex = [item.word, item.index].join('')
```

Comment:
`${name}${key}` use it

Reply:
+

Reply:
I do not agree. ${} is a template syntax, for JSX etc.